### ISSUE TITLE
Missing snapshot class causes panic in release-2.1 branch

### ISSUE THREAD

**Issue by mattcary** [kubernetes-csi/external-snapshotter#445]
#380 is active in release-2.1 (and maybe others).

The fix #381 can be backported.

/assign

**Comment by mattcary** [comment]
I've confirmed that it is not a problem for release-2.0 (the bug was introduced in #275).

I haven't attempted to test 1.x, which uses the alpha API.

I use the following to test: apply it, and then see if the snapshot controller errors, in particular looking for a nil pointer dereference error in the logs. (note it depends on installing a standard-rwo StorageClass)

```
kind: PersistentVolumeClaim
apiVersion: v1
metadata:
  name: busy-pvc
spec:
  accessModes: [ "ReadWriteOnce" ]
  storageClassName: standard-rwo
  resources:
    requests:
      storage: 10Gi

---

apiVersion: v1
kind: Pod
metadata:
  name: busy
spec:
  containers:
   - name: busy
     image: busybox 
     args: ["sleep", "3600"]
     volumeMounts:
       - mountPath: /data/.test
         name: mypvc
  volumes:
   - name: mypvc
     persistentVolumeClaim:
       claimName: busy-pvc
       readOnly: false

---

apiVersion: snapshot.storage.k8s.io/v1beta1
kind: VolumeSnapshot
metadata:
  name: new-snapshot
spec:
  source:
    persistentVolumeClaimName: busy-pvc
```

**Comment by mattcary** [comment]
/close

Fixed with v2.1.3.
